pkg/querier/iterators/chunk_iterator.go

```diff
@@ -34,24 +38,19 @@ func (i *chunkIterator) Seek(t int64) chunkenc.ValueType {
 	// We assume seeks only care about a specific window; if this chunk doesn't
 	// contain samples in that window, we can shortcut.
 	if int64(i.Through) < t {
```

Comment:
Hmm, this problem is in the baseline already, it could in theory return false from Seek and true from a subsequent Next. Creating separate issue: https://github.com/grafana/mimir/issues/4415